compiler-core/src/type_.rs

```diff
@@ -1279,6 +1279,9 @@ pub fn generalise(t: Arc<Type>) -> Arc<Type> {
 pub enum FieldAccessUsage {
     /// Used as `thing.field()`
     MethodCall,
+    /// Used internally when trying to access a field when
+    /// performing record updates
+    RecordUpdate,
```

Comment:
It's not clear to me what this means, just by the name and description I'd guess it's this
```gleam
Wobble(..wobble, field: other.field)
//                            ^^^^^
```
Did you mean access to a field that is inferred to be safe after narrowing? I'd change the name in that case

Reply:
Yeah sorry I'll clarify that